component-library/component-lib/src/lib/form-components/multi-checkbox/multi-checkbox.component.ts

```diff
@@ -34,123 +38,223 @@ export class MultiCheckboxComponent implements OnInit {
-            response.event.remove === true
-          ) {
-            this.removeResponseFromError(response);
-          } else {
-            this.config?.errorMessages.push(response.event);
-          }
-        } else if (
-          this.config.children &&
-          this.config.children?.findIndex((child) => {
-            return child.id === response.id;
-          }) > -1
-        ) {
-          if (
-            response.event.hasOwnProperty('remove') &&
-            response.event.remove === true
-          ) {
-            let newErrorMessages = new Array<IErrorPairsMutltiCheckBox>();
-
-            this.config.errorMessages.forEach((error) => {
-              if (error.id != response.id) {
-                newErrorMessages.push(error);
-              }
-            });
+    if (this.config.parent != undefined) {
+      this.groupCheckbox = false
+      this.configSub =
+        this.multicheckboxService.multiCheckboxEventObs$.subscribe(
+          (response) => {
+            if (this.config.children) {
+              if (response.id === this.config.parent?.id) {
+                this.config.children?.forEach((res) => {
+                  this.multicheckboxService.checkEvent({
+                    id: res.id,
+                    event: response.event
+                  });
+                });
+                this.config.parent.mixed = false;
+
+                this.checkError(
+                  this.config?.parent?.formGroup?.get(this.config.parent?.id)
+                    ?.status,
+                  this.config?.parent?.formGroup || new FormGroup({}),
+                  this.config.parent?.id
+                );
+
+                this.config.children?.forEach((res) => {
+                  this.checkError(
+                    this.config?.parent?.formGroup?.get(res.id)?.status,
+                    this.config?.parent?.formGroup || new FormGroup({}),
+                    res.id
+                  );
+                });
+              } else if (
+                this.config.children?.findIndex((child) => {
+                  return child.id === response.id;
+                }) > -1
+              ) {
+                let positive = 0;
+                let negative = 0;
+
+                this.config.children?.forEach((res) => {
+                  res.formGroup.get(res.id)?.value === true
+                    ? positive++
+                    : negative++;
+                });
+
+                if (positive > 0 && negative > 0) {
+                  this.config.parent?.formGroup
+                    .get(this.config.parent?.id)
+                    ?.patchValue(true, { emitEvent: false });
+                  this.config!.parent!.mixed = true;
+                } else if (positive > 0 && negative == 0) {
+                  this.config.parent?.formGroup
+                    .get(this.config.parent?.id)
+                    ?.patchValue(true, { emitEvent: false });
+
+                  this.config!.parent!.mixed = false;
```

Comment:
Is this inside a null check? If so, the '!' are fine, but if not, please replace with '?'

Reply:
It is under "if(this.config.parent != undefined)"